wgpu-core/src/identity.rs

```diff
@@ -32,98 +35,99 @@ use std::fmt::Debug;
 /// [`alloc`]: IdentityManager::alloc
 /// [`free`]: IdentityManager::free
 #[derive(Debug, Default)]
-pub struct IdentityManager {
-    /// Available index values. If empty, then `epochs.len()` is the next index
-    /// to allocate.
-    free: Vec<Index>,
-
-    /// The next or currently-live epoch value associated with each `Id` index.
-    ///
-    /// If there is a live id with index `i`, then `epochs[i]` is its epoch; any
-    /// id with the same index but an older epoch is dead.
-    ///
-    /// If index `i` is currently unused, `epochs[i]` is the epoch to use in its
-    /// next `Id`.
-    epochs: Vec<Epoch>,
+pub(super) struct IdentityValues {
+    free: Vec<(Index, Epoch)>,
+    //sorted by Index
+    used: HashMap<Epoch, Vec<Index>>,
```

Comment:
This seems to be the only other spot where an std hashmap is used. It might be beneficial to use a faster hasher here as well.